jsonschema-core.md

```diff
@@ -2146,540 +2119,48 @@ a fragment in order to identify the specific schema object.
-          "instanceLocation": "/foo"
-        },
-        {
-          "valid": true,
-          "evaluationPath": "/properties/foo/allOf/1",
-          "schemaLocation": "https://json-schema.org/schemas/example#/properties/foo/allOf/1",
-          "instanceLocation": "/foo",
-          "annotations": {
-            "title": "foo-title",
-            "properties": [
-              "foo-prop"
-            ],
-            "additionalProperties": [
-              "unspecified-prop"
-            ]
-          },
-          "details": [
-            {
-              "valid": true,
-              "evaluationPath": "/properties/foo/allOf/1/properties/foo-prop",
-              "schemaLocation": "https://json-schema.org/schemas/example#/properties/foo/allOf/1/properties/foo-prop",
-              "instanceLocation": "/foo/foo-prop",
-              "annotations": {
-                "title": "foo-prop-title"
-              }
-            },
-            {
-              "valid": true,
-              "evaluationPath": "/properties/foo/allOf/1/additionalProperties",
-              "schemaLocation": "https://json-schema.org/schemas/example#/properties/foo/allOf/1/additionalProperties",
-              "instanceLocation": "/foo/unspecified-prop"
-            }
-          ]
-        }
-      ]
-    },
-    {
-      "valid": true,
-      "evaluationPath": "/properties/bar",
-      "schemaLocation": "https://json-schema.org/schemas/example#/properties/bar",
-      "instanceLocation": "/bar",
-      "details": [
-        {
-          "valid": true,
-          "evaluationPath": "/properties/bar/$ref",
-          "schemaLocation": "https://json-schema.org/schemas/example#/$defs/bar",
-          "instanceLocation": "/bar",
-          "annotations": {
-            "title": "bar-title",
-            "properties": [
-              "bar-prop"
-            ]
-          },
-          "details": [
-            {
-              "valid": true,
-              "evaluationPath": "/properties/bar/$ref/properties/bar-prop",
-              "schemaLocation": "https://json-schema.org/schemas/example#/$defs/bar/properties/bar-prop",
-              "instanceLocation": "/bar/bar-prop",
-              "annotations": {
-                "title": "bar-prop-title"
-              }
-            }
-          ]
-        }
-      ]
-    }
-  ]
-}
-```
+A dropped annotation is any annotation produced and subsequently dropped by the
+evaluation due to an unsuccessful validation result of the containing subschema.
+This information MAY be included if the validation result of the containing
+subschema was unsuccessful. It MUST NOT be included if the local validation
+result of the containing subschema was successful.
 
-#### Output validation schemas
+Implementations that wish to provide dropped annotations MUST NOT provide them
+as their default behavior. Dropped annotations MUST only be included when the
+implementations is explicitly configured to do so.
```

Comment:
IMO, the spec is overstepping here. I think it's perfectly fine to allow an implementation to make this choice themselves.

Reply:
I see your point, but I still want to emphasize that this data point is generally intended for debugging purposes, and making it available in a production scenario will likely lead to confusion.

Reply:
(For a separate conversaion, but worth mentioning here.)

Moreover, if we drop annotations as an driving mechanism for evaluation, many of the keywords which now produce annotations won't need to.  For example, if `additionalProperties` just looks at the adjacent `properties` keyword's keys to know what remains to evaluate, then `properties` doesn't need to emit its annotation.

That said, I'd probably still track that data internally and treat it as a sort of "internal annotation" in order to avoid duplicate effort like having to re-analyze the `patternProperties` keys against an instance's keys.  But that doesn't mean that it needs to be exposed.

I still have to do some research here to find out what annotations are being used.

Reply:
In terms of the highlighted content, I feel like SHOULD NOT and SHOULD would be enough here. There may be implementations that are specifically designed to do that by default, for example.

@gregsdennis to your comment, I don't think there's any harm in saying explicitly that this sort of behaviour is intended to aid debugging in non-production environments. Of course, that could mean for developer tooling, such a thing is wanted by default.